packages/TelegramClient-UI.package/TCUEmojiHelper.class/class/add.to..st

```diff
@@ -0,0 +1,9 @@
+unicode matching
+add: aCharacter to: aRunCollection
+
+	(aCharacter isOctetCharacter) ifTrue: [
+		aRunCollection add: aCharacter.
+	] ifFalse: [
+		self addSingle: aCharacter codePoint printStringHex to: aRunCollection.
```

Comment:
The problem is, that we have not found a way of decoding *one* emoji on its own. Take for example the `kiss: woman, woman` emoji 👩‍❤️‍💋‍👩. It is encoded as `1F469-200D-2764-FE0F-200D-1F48B-200D-1F469`, `Woman`, `ZWJ`, `Red Heart`, `VS16`. `ZWJ`, `Kiss`, `ZWJ` and `Woman`. One could think, that we simply check for `VS16` and `ZWJ` to find the whole sequence. However, the `woman running: medium skin tone` emoji, 🏃🏽‍♀️, is encoded as `1F3C3-1F3FD-200D-2640-FE0F`. There is another `medium skin tone` character after the `person running` character. 
On top of that, different clients and operating systems send the same character in different ways. To mitigate that, we remove all characters like the `VS16` character from our filenames and from the incoming text and match them by taking the full sequence, all but last, all but two last, etc. to find the longest match.

Reply:
the `runCollection` is a collection of characters and collections containing unicode hex strings. We do it this way to not have to care about normal characters when decoding emojis

Reply:
Okay we understand and it looks like a pretty difficult thing...
However, we think it would be good to rename `runCollection` to something like just `charsAndEmojis`?
We think the main problem in our view here is the non-descriptive name "`run..`" which it hard to understand and very generic. We have some suggestions in the following methods for other names. Maybe you can take that into consideration ;).

Reply:
Will do, I agree, the name "run" is not very descriptive